### ISSUE TITLE
Combo Box removable options should have clear instructions conveyed via accessible text of the button

### ISSUE THREAD

**Issue by jake-costa** [jpmorganchase/salt-ds#5755]
### Latest version

- [x] I have tested the latest version

### Description

The current combo box multi-select example are using a button in addition to the accessible text conveyed via the close icon. In addition, because of this, screen reader reliant users are hearing close instead of remove for the selected item.

### Steps to reproduce

.

### Expected behavior

Ideally, the button itself should have the accessible name and removal instructions and the close icon should be hidden

### Package name(s)

Core (@salt-ds/core)

### Browser

- [x] Chrome
- [x] Safari
- [x] Firefox
- [ ] Microsoft Edge

### Operating system

- [x] macOS
- [x] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Are you a JPMorgan Chase & Co. employee?

- [x] I am an employee of JPMorgan Chase & Co.
